### DISCUSSION TITLE
Multithreaded processes in AWS Batch environment

### DISCUSSION THREAD

**Discussion by askol-lurie** [nextflow-io/nextflow#2780] [Q&A] [answered]
Hi Folks,
  I was curious if there was a trick for having nextflow specify the number of vcpus in the AWS Batch job description when running a process (see below) that allow for multithreading (e.g. BWA). Let's assume that the Batch environment contains instances that support that number of vcpus necessary for the amount of multithreading.  My impression is that it doesn't do this naturally based on the fact that I checked in on a job running BWA an it was assigned a single vcpu.  Any help would be greatly appreciated.

```

process bwa {
        publishDir "$params.outdir/bam", mode: 'copy'

        //take input from trim_fastq and run bwa mem
	tag "$sample"

        input:
                //tuple val(sample), file(reads) 
                tuple val(sample), path(reads) 
                path(x)

        output:
                tuple val(sample), path("${sample}.trim.align.bam"), emit: bam_ch

        script:
        """
                bwa mem -aM -R "@RG\\tID:${sample}\\tSM:${sample}\\tPL:ILLUMINA\\tPI:330" -t 4 ${x[1]} ${reads[0]} ${reads[1]} | samtools view -Sb -o ${sample}.trim.align.bam
        """
}

```
Best, 

Andrew

**Comment by pditommaso** [comment, marked as answer]
You need to use the usual [cpus](https://www.nextflow.io/docs/latest/process.html#cpus) directive in the process definition

**Reply by askol-lurie** [reply]
Thank you. That seem pretty obvious.
